### ISSUE TITLE
containerd listens on port 10010 on public interface

### ISSUE THREAD

**Issue by mimmus** [coreos/bugs#2524]
# Issue Report #
On stable Container Linux release, 'containerd' service listens on port 10010 on public interface and cannot be modified/disabled.
This port could be used from other containers or abused from the network.

## Bug ##
Related to this:
https://github.com/moby/moby/issues/37507
https://github.com/containerd/containerd/issues/2483#issuecomment-437025011

### Container Linux Version ###
Latest (1911) and previous (1855) stable Container Linux release.
As Docker version is the same, probably issue is present also on Beta and Alpha.

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1855.5.0
VERSION_ID=1855.5.0
BUILD_ID=2018-10-22-2305
PRETTY_NAME="Container Linux by CoreOS 1855.5.0 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

### Environment ###

N/A

### Expected Behavior ###
Service should be disabled or listen on localhost

### Actual Behavior ###
Service listens on public IP and cannot be disabled

### Reproduction Steps ###

  1. Fresh install of Container Linux
  2. Look at:
```
$ sudo netstat -anp | grep 10010
tcp        0      0 10.57.0.236:10010       0.0.0.0:*               LISTEN      871/containerd
```
### Other Information ###
Bug should be fixed in Docker 18.06.1 but it is still present on Container Linux

**Comment by mimmus** [comment]
Thanks for your consideration.
As a workaround (I need to wait for certification of my application stack before upgrading Container Linux), is there a containerd unit that I can disable?

**Comment by dm0-** [comment]
You could write [the new config file](https://github.com/coreos/coreos-overlay/blob/master/app-emulation/containerd/files/config.toml) somewhere, then write a `containerd.service` drop-in that overrides the environment variable `CONTAINERD_CONFIG` with its path.

**Comment by bgilbert** [comment]
This should be fixed in alpha 1967.0.0 and beta 1939.2.1, due shortly.

**Comment by bgilbert** [comment]
This should be fixed in stable 1911.4.0, due shortly.  Thanks again for reporting.
